### ISSUE TITLE
add terminal icon support

### ISSUE THREAD

**Issue by zejafo** [EthanSK/restore-terminals-vscode#10]
After the latest vscode-insiders builds the team add support of terminal tabs and add icons.
It would be great if you add icon parameter to the configuration of restoreTerminals.

![image](https://user-images.githubusercontent.com/7479125/116254325-ec2a2b00-a779-11eb-9517-90559eda9496.png)

**Comment by bferg317-perrigo** [comment]
It's not critical obviously, but it would be nice to use icons and acronyms to shrink my tabs pane width-wise.

**Comment by EthanSK** [comment]
Any ideas on how to do this? Maybe help me find some VSCode APIs for this?

**Comment by 0xktz** [comment]
Well honestly I came here looking about this as well, I'd love having that as well cause especially in situations like these 
![image](https://user-images.githubusercontent.com/20067282/121543671-c5fed880-ca11-11eb-9225-166dbebc8030.png)
where you need to minimize it as much possible due to spacing constrains would definitely help a lot!

Great extension by the way, thanks :)

**Comment by tingspain** [comment]
First of all, super nice extension. I love it! 

Yes I came here looking for the same,  a way to setup the icon. Guys, anyone know if it is possible to set the color? I have been trying for a while but... nothing 

Thanks!

**Comment by Olcatsy** [comment]
Setting the colour would be super useful for sure!

**Comment by rbalet** [comment]
A workaround could be to do this through the cli, but no Idea if this is doable. 
Did open a question on stack overflow. https://stackoverflow.com/questions/68297141/change-vscode-terminal-tabs-icon-and-color-with-the-cli  
I'll keep you inform in case I get any answer.

**Comment by rbalet** [comment]
Seems like vs-code finally let us implement it.

> ## June 2021 (version 1.58)
>Set the icon for terminals created via the extension API
>window.createTerminal now accepts an iconPath, associating an icon with the terminal in its tab.
>
>const term = vscode.window.createTerminal({
>  name: `Serve`,
>  iconPath: new vscode.ThemeIcon('server-process')
>});
>term.show();
>
>![image](https://user-images.githubusercontent.com/44493964/126268458-63a49979-3f4f-4f25-928e-e09c3b838ff3.png)

**Comment by hces10** [comment]
when someone implement icon support, please add terminal color support too, i'm adding the suggestion here because change color and icon is on the same place and it seems easy to add the color support.
![image](https://user-images.githubusercontent.com/42579291/126813579-635a2fd2-8ed6-4840-896c-e7a959b73931.png)

**Comment by laneschmidt** [comment]
@rbalet you're right about manually setting an icon, but vscode's `ExtensionTerminalOptions` does not yet support `icon` or `color` as an input argument at the time of writing (Type Definition for Visual Studio Code 1.46 Extension API):

```
Object literal may only specify known properties, and 'icon' does not exist in type 'ExtensionTerminalOptions'
```

**Comment by rbalet** [comment]
@laneschmidt I'm a bit a newbies and that vsCode API, but does it solve the problem ? https://github.com/microsoft/vscode/issues/128228 -> They did merge it

**Comment by pauIbanez** [comment]
Workaround https://stackoverflow.com/questions/68297141/change-vscode-terminal-tabs-icon-and-color-with-the-cli/71482047#71482047

**Comment by iRoachie** [comment]
@tingspain @Olcatsy Added support for terminal color here https://github.com/EthanSK/restore-terminals-vscode/pull/28
